syntax/forth.vim

```diff
@@ -181,36 +180,28 @@ syn keyword forthBlocks BLOCK-INCLUDED BLK
 " numbers
 syn keyword forthMath DECIMAL HEX BASE
 syn match forthInteger '\<-\=[0-9]\+.\=\>'
-syn match forthInteger '\<&-\=[0-9]\+.\=\>'
```

Comment:
`&123` seems to be Gforth legacy. Since `#123` is standard and works just as well in Gforth, I think it is fine to remove this by now.